File: Week_02/Practicum/Contains Duplicate II.cpp
class Solution {
public:
    bool containsNearbyDuplicate(vector<int>& nums, int k) {
        vector<pair<int, int>> p;

        for(int i = 0; i < nums.size(); i++){
            p.push_back({nums[i], i});
        }

        sort(p.begin(), p.end());

        for(int i = 0; i < nums.size() - 1; i++){
            if (p[i].first == p[i+1].first && abs(p[i].second - p[i+1].second) <= k)
                return true;
        }

        return false;
    }
};

File: Week_02/Practicum/Python/Contains Duplicate II.py
from collections import defaultdict
class Solution:
    def containsNearbyDuplicate(self, nums: List[int], k: int) -> bool:
        nums = [(num, i) for i, num in enumerate(nums)]

        nums.sort()

        for i in range(len(nums) - 1):
            if nums[i][0] == nums[i + 1][0] and nums[i + 1][1] - nums[i][1] <= k:
                return True

        return False